src/menhir_parser.mly

```diff
@@ -59,31 +60,30 @@ let translate_old_rule : old_p_rule -> p_rule = fun r ->
             begin
               try
                 let n = Hashtbl.find arity x in
-                if nb_args < n then Hashtbl.replace arity x nb_args
+                if nb_args > n then Hashtbl.replace arity x nb_args
               with Not_found -> Hashtbl.add arity x nb_args
             end
       | P_Wild          -> ()
-      | P_Type          -> ()
```

Comment:
That change should be reverted, that error should be caught at scoping time (which it is).

Reply:
Why this one should be caught at scoping time and not P_Prod?

Reply:
Well, it would be best to also handle `P_Prod` at scoping time, but in that case the code for the translation is annoying to write. In any case, my main point was that these changes are not necessary.